universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -326,6 +326,32 @@ private void putParentArray(Path path) {
     }
   }
 
+  /**
+   * Reads the value at the given path as a string. Returns {@link Optional#empty} if there is no
+   * value at the path. For JSON arrays of strings, this formats the array as a string according to
+   * {@link ImmutableList#toString}.
+   *
+   * @param path the {@link Path} to read
+   * @return optionally returns the value at the path as a string if it exists, or empty if not
+   */
+  public Optional<String> readAsString(Path path) {
+    if (isJsonArray(path)) {
+      return readList(path).map(ImmutableList::toString);
```

Comment:
(nonblocking) could this not always map `toString`?

Reply:
Ahh i see in the tests it packs numerical values into strings.

Reply:
Yeah every other type is fine - it just doesn't understand JSON arrays or objects (but  we don't have objects yet)

Thanks for the review!